### PR TITLE
Profile Type filtering inspector fix

### PR DESCRIPTION
**Pull request by Troy-Ferrell**
## Overview
Two primary bugs here with profile inspectors

1) In RenderProfileInternal, the call to `MixedRealityProfileUtility.GetProfileTypesForService()` was incorrectly utitlized by selecting FirstOrDefault from the available list even if that profile was not a match for the given service type. Further, this call was made even if the profileType argument was already not null

2) Some inspectors were not properly providing a service type to RenderProfile(). They made erroneous checks before setting the serviceType argument.

With this change, now profile drop downs are actually filtered (where applicable like Spatial Mesh Observers) and correct value can now be set unblocking bug referenced below

## Changes
- Fixes: #6294

### CODE REVIEW

File: Assets/MixedRealityToolkit/Inspectors/Profiles/MixedRealityInputSystemProfileInspector.cs [hunk 1]
```diff
@@ -273,11 +273,7 @@ private void RenderList(SerializedProperty list)
...
 
-                                System.Type serviceType = null;
-                                if (configurationProfile.objectReferenceValue != null)
-                                {
-                                    serviceType = (target as MixedRealityInputSystemProfile).DataProviderConfigurations[i].ComponentType;
-                                }
+                                System.Type serviceType = (target as MixedRealityInputSystemProfile).DataProviderConfigurations[i].ComponentType;
 
                                 changed |= RenderProfile(configurationProfile, null, true, false, serviceType);
                             }
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit/Inspectors/Profiles/MixedRealityInputSystemProfileInspector.cs]: You may need to add the typeof() check as in spatial

Comment (Troy-Ferrell) [reply Assets/MixedRealityToolkit/Inspectors/Profiles/MixedRealityInputSystemProfileInspector.cs]: No actually, the profile type under DataProviderConfigurations is actually BaseMixedRealityProfile
image

While the ObserverConfigurations in SA, explicitly use BaseSpatialAwarenessObserverProfile

image

**Pull request merged**